docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,524 @@
+# Writing a simple publisher and subscriber (RUST)
+* Goal: Create and run a publisher and subscriber node using Python.
+* Tutorial level: Beginner
+* Time: 20 minutes
+<details><summary>Background</summary>
+
+In this tutorial you will create a pair of 
+[nodes](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Nodes/Understanding-ROS2-Nodes.html) that pass information to each other via a 
+[topic](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Topics/Understanding-ROS2-Topics.html) in the form of string messages. The example used here is a simple "talker" and "listener" system; one node publishes data and the other subscribes to the topic to receive that data.
+
+Since Rust doesn't have inheritance, it's not possible to inherit from `Node` as is common practice in [`rclcpp`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html) or [`rclpy`](https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html).
+
+The code used in these examples can be found [here](https://gitlab.com/ros21923912/simple_ros2_node)  
+<div style="margin-left:20px;">
+<details><summary>Side-note on dependencies</summary>
+
+You may be wondering why you can't just add all your ros2-specific dependencies to `cargo.toml` with `cargo add ${dependencie}` and have to edit this file manually. Here is why:
+Almost none of the ROS2 dependencies you'll need for your ros2 rust node development currently exist on [crates.io](https://crates.io/), the main source for rust depencies. So the add command simply can't find the dependency targets. What colcon does by compiling the ros2 rust dependencies and your ros2 rust project is redirect the cargo search for dependencies directly into your `workspace/install` folder, where it'll find locally generated rust projects to use as dependencies. In particular, almost all message types will be called as dependencies for your ros2 rust project this way.
```

Comment:
```suggestion
Almost none of the ROS 2 dependencies you'll need for your ROS 2 Rust node development currently exist on [crates.io](https://crates.io/), the main source for rust depencies. So the add command simply can't find the dependency targets. What colcon does by compiling the ROS 2 Rust dependencies and your ROS 2 Rust project is redirect the cargo search for dependencies directly into your `workspace/install` folder, where it'll find locally generated rust projects to use as dependencies. In particular, almost all message types will be called as dependencies for your ROS 2 Rust project this way.
```

The correct spelling of ROS 2 is separately (not ROS2). It's not clear to me what you meant with ros2 rust, is it `ros2-rust` (this project) or ROS 2 Rust?